### ISSUE TITLE
Add a type for decimal numbers

### ISSUE THREAD

**Issue by hvbtup** [nanosai/rion-ops-java#1]
A decimal type would be useful.
Decimal numbers cannot be serialized as float/double as the conversion will cause precision loss / rounding issues.
As of now, one has to convert decimal numbers to Strings and then use the UTF-8 type. This circumvents conversion issues, but it has a few drawbacks:
- Waste of band-width, because each decimal digit takes one byte.
- Loss of semantics (for example, a generic viewer program for files of RION objects could use highlighting to show numbers in a different color than strings if we had a decimal type.

Thus I think it could be useful to officially add a type for decimal numbers which are encoded as in 
https://en.wikipedia.org/wiki/Densely_packed_decimal

**Comment by jjenkov** [comment]
Interesting! I assume you mean arbitrary precision decimal numbers, right? Like e.g. Java's BigDecimal ?

**Comment by hvbtup** [comment]
Well, more or less.
The best Java pendant would be BigDecimal.
But I think nobody (except mad scientists) needs arbitrary precision. decimal64 should be good enough. It supports 16 significant digits. That's enough for monetary values and enough for the LIMS environment (which is my focus).

The decimal64 in DPD format look as bit complicated (basically, 10 bits represent 3 decimal digits). 

But an implementation for encoding and decoding already exists, see https://github.com/FirebirdSQL/decimal-java 

Note that Apache Commons Math decimal64 means something completely different (and they did it wrong, because the constructor uses a double as input and thus causes conversion issues).

**Comment by jjenkov** [comment]
So, a 64 bit float (double) won't work, right? That is what causes the conversion errors (rounding /cutting / approximation of decimals) ?

**Comment by hvbtup** [comment]
Yes. That's the whole reason why classes like BigDecimal etc exist: To avoid conversion errors when the end result needs to be presented to a human. This is particularly important in the finance world.
